QUARKUS-2787 Rest Data Panache: Correct Open API integration (#125)
quarkus-qe/quarkus-test-plans: QUARKUS-2787.md

Affected extensions:
- `quarkus-hibernate-orm-rest-data-panache`
- `quarkus-hibernate-reactive-rest-data-panache` (experimental)
- `quarkus-mongodb-rest-data-panache` (experimental)
- `quarkus-spring-data-rest` - the REST Data Panache feature is not documented in the extension's guide

### Existing tests
There are integration tests in upstream for:
- `quarkus-hibernate-orm-rest-data-panache` - https://github.com/quarkusio/quarkus/blob/main/extensions/panache/hibernate-orm-rest-data-panache/deployment/src/test/java/io/quarkus/hibernate/orm/rest/data/panache/deployment/openapi/OpenApiIntegrationTest.java
- `quarkus-hibernate-reactive-rest-data-panache` - https://github.com/quarkusio/quarkus/blob/main/extensions/panache/hibernate-reactive-rest-data-panache/deployment/src/test/java/io/quarkus/hibernate/reactive/rest/data/panache/deployment/openapi/OpenApiIntegrationTest.java

They provide sufficient coverage of the OpenAPI functionality for the extensions.

## Automated test development
Adopt the upstream integration tests for extensions not yet covered:
- `quarkus-spring-data-rest` - module `spring/spring-data`

## Advanced topics for test development

> - `quarkus-mongodb-rest-data-panache` - nice to have. No test coverage for this extension is present in Quarkus QE TS yet.

Reviewer: This extension is not supported on product side, so no need to cover it atm.

```suggestion
- `quarkus-mongodb-rest-data-panache` - for the future, this extension is not supported on product side yet.
```

## Impact on test suites and test environment
1 new scenario, expected increase in the execution time:

> 2 / 5 / 4 / 8 minutes in JVM / native / OpenShift JVM / OpenShift native mode.

Reviewer: Are there any OpenShift specifics that require OpenShift JVM and Native runs?

Author: Not that I know of, I just thought it would be good to have an OpenShift scenario for OpenAPI + Rest Data Panache. I couldn't find one anywhere.
